### DISCUSSION TITLE
how to search in multiple directories with specified extensions

### DISCUSSION THREAD

**Discussion by xinglonghe** [BurntSushi/ripgrep#1607] [General] [answered]
windows findstr can do this like:
findstr /PRISN /D:root\dir1;root\dir2 /C:foo *.c *.h

which search in two directories "root\dir1" and "root\dir2", and only search the files with extension "*.c" and "*.h".

I know that ripgrep can do this with:
rg foo -g root/dir1/*.c -g root/dir1/*.h -g root/dir2/*.c -g root/dir2/*.h

the command looks too tedious, especially if we want to specify too many directories and extensions.

Is there a simpler syntax for ripgrep doing this?

**Comment by BurntSushi** [comment, marked as answer]
`rg root/dir1 root/dir2 -tc foo`

**Reply by xinglonghe** [reply]
I see,  thanks. so just use "-g" for specific extensions:
rg foo root/dir1 root/dir2 -g '*.{c,h}'

**Reply by BurntSushi** [reply]
Sure, that works too.
